File: ALODWebUtility/TabNavigator/TabItem.vb
Imports ALOD.Core.Domain.Workflow

Namespace TabNavigation

    <Serializable()>
    Public Class TabItem

        Dim _access As PageAccessType = PageAccessType.None
        Dim _active As Boolean = False
        Dim _completed As Boolean = False
        Dim _enabled As Boolean = False
        Dim _order As Short = 0
        Dim _page As String = String.Empty
        Dim _readOnly As Boolean = True
        Dim _required As Boolean = False
        Dim _script As String = String.Empty
        Dim _title As String = String.Empty
        Dim _visible As Boolean = True

        Public Sub New(ByVal title As String, ByVal order As Short, ByVal page As String)
            _title = title
            _order = order
            _page = page
        End Sub

        Public Sub New(ByVal title As String)
            _title = title
        End Sub

        Public Property Access() As PageAccessType
            Get
                Return _access
            End Get
            Set(ByVal value As PageAccessType)
                _access = value
            End Set
        End Property

        Public Property Active() As Boolean
            Get
                Return _active
            End Get
            Set(ByVal Value As Boolean)
                _active = Value
            End Set
        End Property

        Public Property ClientScript() As String
            Get
                Return _script
            End Get
            Set(ByVal value As String)
                _script = value
            End Set
        End Property

        Public Property Completed() As Boolean
            Get
                Return _completed
            End Get
            Set(ByVal Value As Boolean)
                _completed = Value
            End Set
        End Property

        Public Property Enabled() As Boolean
            Get
                Return _enabled
            End Get
            Set(ByVal Value As Boolean)
                _enabled = Value
            End Set
        End Property

        Public Property IsReadOnly() As Boolean
            Get
                Return _readOnly
            End Get
            Set(ByVal value As Boolean)
                _readOnly = value
            End Set
        End Property

        Public ReadOnly Property Order() As Short
            Get
                Return _order
            End Get
        End Property

        Public ReadOnly Property Page() As String
            Get
                Return _page
            End Get
        End Property

        Public Property Required() As Boolean
            Get
                Return _required
            End Get
            Set(ByVal value As Boolean)
                _required = value
            End Set
        End Property

        Public Property Title() As String
            Get
                Return _title
            End Get
            Set(value As String)
                _title = value
            End Set
        End Property

        Public Property Visible() As Boolean
            Get
                Return _visible
            End Get
            Set(ByVal value As Boolean)
                If (value = True AndAlso _access = PageAccessType.None) Then
                    Exit Property 'don't allow tabs with no access to become visible
                End If
                _visible = value
            End Set
        End Property

    End Class

End Namespace

File: ALODWebUtility/TabNavigator/TabItem.cs
using System;
using ALOD.Core.Domain.Workflow;

namespace ALODWebUtility.TabNavigation
{
    [Serializable]
    public class TabItem
    {
        PageAccessType _access = PageAccessType.None;
        bool _active = false;
        bool _completed = false;
        bool _enabled = false;
        short _order = 0;
        string _page = string.Empty;
        bool _readOnly = true;
        bool _required = false;
        string _script = string.Empty;
        string _title = string.Empty;
        bool _visible = true;

        public TabItem(string title, short order, string page)
        {
            _title = title;
            _order = order;
            _page = page;
        }

        public TabItem(string title)
        {
            _title = title;
        }

        public PageAccessType Access
        {
            get
            {
                return _access;
            }
            set
            {
                _access = value;
            }
        }

        public bool Active
        {
            get
            {
                return _active;
            }
            set
            {
                _active = value;
            }
        }

        public string ClientScript
        {
            get
            {
                return _script;
            }
            set
            {
                _script = value;
            }
        }

        public bool Completed
        {
            get
            {
                return _completed;
            }
            set
            {
                _completed = value;
            }
        }

        public bool Enabled
        {
            get
            {
                return _enabled;
            }
            set
            {
                _enabled = value;
            }
        }

        public bool IsReadOnly
        {
            get
            {
                return _readOnly;
            }
            set
            {
                _readOnly = value;
            }
        }

        public short Order
        {
            get
            {
                return _order;
            }
        }

        public string Page
        {
            get
            {
                return _page;
            }
        }

        public bool Required
        {
            get
            {
                return _required;
            }
            set
            {
                _required = value;
            }
        }

        public string Title
        {
            get
            {
                return _title;
            }
            set
            {
                _title = value;
            }
        }

        public bool Visible
        {
            get
            {
                return _visible;
            }
            set
            {
                if (value == true && _access == PageAccessType.None)
                {
                    return; // don't allow tabs with no access to become visible
                }
                _visible = value;
            }
        }
    }
}
